packages/cardano-services/src/Asset/DbSyncAssetProvider.ts

```diff
@@ -0,0 +1,89 @@
+import { Asset, AssetProvider, Cardano, ProviderError, ProviderFailure } from '@cardano-sdk/core';
+import { AssetBuilder } from './AssetBuilder';
+import { DbSyncProvider } from '../DbSyncProvider';
+import { Logger } from 'ts-log';
+import { NftMetadataService, TokenMetadataService } from './types';
+import { Pool } from 'pg';
+
+/**
+ * Dependencies that are need to create DbSyncAssetProvider
+ */
+export type DbSyncAssetProviderDependencies = {
+  /**
+   * The db-sync database Pgpool
+   */
+  db: Pool;
+
+  /**
+   *
+   * The logger object
+   */
+  logger: Logger;
+
+  /**
+   * The NftMetadataService to retrieve Asset.NftMetadata.
+   */
+  ntfMetadataService: NftMetadataService;
+
+  /**
+   * The TokenMetadataService toretrieve Asset.TokenMetadata.
+   */
+  tokenMetadataService: TokenMetadataService;
+};
+
+type AssetExtraData = Parameters<AssetProvider['getAsset']>[1];
+
+/**
+ * AssetProvider implementation using NftMetadataService, TokenMetadataService
+ * and cardano-db-sync database as sources
+ */
+export class DbSyncAssetProvider extends DbSyncProvider implements AssetProvider {
+  #builder: AssetBuilder;
+  #dependencies: DbSyncAssetProviderDependencies;
+
+  constructor(dependencies: DbSyncAssetProviderDependencies) {
+    const { db, logger } = dependencies;
+
+    super(db);
+
+    this.#builder = new AssetBuilder(db, logger);
+    this.#dependencies = dependencies;
+  }
+
+  async close() {
+    this.#dependencies.tokenMetadataService.shutdown();
```

Comment:
Since it's injected in the constructor and isn't started by DbSyncAssetProvider, it should probably not be shutting it down

Reply:
Resolved in c1c4c55d8